Userland/Libraries/LibGfx/ImageFormats/AnimationWriter.h

```diff
@@ -23,6 +23,8 @@ class AnimationWriter {
     virtual ErrorOr<void> add_frame(Bitmap&, int duration_ms, IntPoint at = {}) = 0;
 
     ErrorOr<void> add_frame_relative_to_last_frame(Bitmap&, int duration_ms, RefPtr<Bitmap> last_frame);
+
+    virtual ErrorOr<void> finalize() { return {}; }
```

Comment:
Is the trailer very large?

I intentionally didn't give this a finalize() method, with the intent that the writers always write a valid file when a frame is added, so that if you interrupt writing in the middle, you still have a valid file.

Maybe we could write the trailer after every frame and then seek and write on top of it when the next frame comes and write a new trailer, and so on?

Reply:
It's one byte, so yeah that's a better solution, thanks!